src/components/inputs/TextField.js

```diff
@@ -175,6 +175,10 @@ export default class TextField extends BaseInput {
     if (prevState.value !== this.state.value || prevProps.focused !== this.state.focused) {
       this.updateFloatingPlaceholderState();
     }
+
+    if (prevProps.value !== this.props.value) {
```

Comment:
Unsure if this should be here or in `BaseInput`

Reply:
why do we need it at all?

Reply:
If we do not add this the error is not cleared when the selection is changed in the Picker. Since validation only occurs onChangeText until now, a prop change does not affect it.

Reply:
the question is - is it an issue? 
If we aim to have a controlled component that means the user is responsible for clearing the error by passing undefined to `error` prop when the value has changed.

Reply:
I actually now looked at the related private PR.
I think I get where you're going with this. 
Let's take this offline.

Reply:
Done